typescript/commands/references.py

```diff
@@ -173,3 +191,42 @@ def run(self, text, argsJson):
         # serialize the reference info into the settings
         self.view.settings().set('refinfo', ref_info.as_value())
         self.view.set_read_only(True)
+
+class RefsHelper:
+
+    __lines = {}
+    __content = {}
+
+    def lineText(self, fileName, line):
+        """ returns text of the given line in the given file """
+        lines = self.__table(fileName)
+        l = len(lines)
+        if l < line:
+            return ''
+        content = self.__content[fileName]
+        start = lines[line]
+        if line < l - 1:
+            end = lines[line + 1]
+        else:
+            end = len(content)
+        return content[start:end].rstrip()
+
+    def __table(self, fileName):
+        """ returns line table for a given filename, computes if needed """
+        lines = self.__lines.get(fileName)
+        if lines is not None:
+            return lines
+        f = open(fileName)
+        lines = []
```

Comment:
Wouldn't it be better to collect the lines as we read them out of the file? With this way, we are reading the entire file into memory, whereas if we iterate one line at a time and toss the lines when we don't need them, we only ever have a line in memory